From 018ec89d460f8b6389b4ffe787d293090dcc0bdd Mon Sep 17 00:00:00 2001
From: Hamster Tian <haotia@gmail.com>
Date: Thu, 30 Jun 2016 13:44:54 +0800
Subject: [PATCH] Nfc: read extra properties of Mifare Classic

* Fixes NFC TagInfo crash with MFC and native NFC stack.

Change-Id: Ib5042f6969d34070fbf0b54b26596b86eda1cd26
---
 nci/src/com/android/nfc/dhimpl/NativeNfcTag.java | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/nci/src/com/android/nfc/dhimpl/NativeNfcTag.java b/nci/src/com/android/nfc/dhimpl/NativeNfcTag.java
index ac7d6b8..fb8b145 100755
--- a/nci/src/com/android/nfc/dhimpl/NativeNfcTag.java
+++ b/nci/src/com/android/nfc/dhimpl/NativeNfcTag.java
@@ -715,6 +715,17 @@ private boolean isUltralightC() {
                         break;
                     }
 
+                    case TagTechnology.MIFARE_CLASSIC: {
+                        byte[] actBytes = mTechActBytes[i];
+                        if ((actBytes != null) && (actBytes.length > 0)) {
+                            extras.putShort(NfcA.EXTRA_SAK, (short) (actBytes[0] & (short) 0xFF));
+                        } else {
+                            // ignore this case.
+                        }
+                        extras.putByteArray(NfcA.EXTRA_ATQA, mTechPollBytes[i]);
+                        break;
+                    }
+
                     case TagTechnology.NFC_BARCODE: {
                         // hard code this for now, this is the only valid type
                         extras.putInt(NfcBarcode.EXTRA_BARCODE_TYPE, NfcBarcode.TYPE_KOVIO);
